app/components/layout/Flashes/index.js

```diff
@@ -16,18 +20,77 @@ class Flashes extends React.Component {
   }
 
   componentDidMount() {
-    FlashesStore.on("flash", this.handleStoreChange);
+    FlashesStore.on('flash', this.handleStoreChange);
+    PusherStore.on('unavailable', this.handleConnectionError);
+    PusherStore.on('connected', this.handleConnectionSuccess);
   }
 
   componentWillUnmount() {
-    FlashesStore.off("flash", this.handleStoreChange);
+    FlashesStore.off('flash', this.handleStoreChange);
+    PusherStore.off('unavailable', this.handleConnectionError);
+    PusherStore.off('connected', this.handleConnectionSuccess);
+  }
+
+  handleStoreChange = (payload) => {
+    this.setState({ flashes: this.state.flashes.concat(payload) });
+  };
+
+  hasConnectionErrorFlash() {
+    return this.state.flashes.length > 0 && this.state.flashes[0].id === FLASH_CONN_ERROR_ID;
   }
 
+  // show a flash when there's a push connection issue
+  handleConnectionError = () => {
```

Comment:
Without knowing too much about when pusher sends the `unavailable` call, but perhaps we should maybe only show the flash if we've been unavailable for more than 10 seconds or so? Maybe more than 60 seconds? I just fear that Australia's shitty internet connect will mean that we're forever seeing that message.